010-server-side-rendering.md

```diff
@@ -0,0 +1,205 @@
+# Server-Side Rendering
+
+We should use [Next.js](https://nextjs.org) and [Heroku](https://www.heroku.com) to power server-side rendering for Phoenix.
+
+**Discussion:** https://github.com/DoSomething/rfcs/pull/10
+
+## Problem
+
+Our front-end application, Phoenix, is not as scalable, simple, or performant as we want:
+
+- Because Phoenix runs on traditional server processes, our **ability to scale is limited** (for authenticated users). If we’re unable to gracefully scale with traffic spikes, we might lose engagement at a key time.
+- The application is split between a traditional backend server, written in [Laravel](https://laravel.com), and a client-side application, written in [React](https://reactjs.org). This **increases context switching for developers**, slowing down development.
+- Because the application is almost completely rendered on the client, we see **reduced performance on slower devices** (like old phones). This may reduce engagement, since nobody likes waiting.
+- Finally, we've been limited in what we can share on social networks (whose scrapers do not execute client-side code) since any social metadata must be fetched and rendered by a separate server-side process.
+
+Despite these problems, we're generally happy with what we've built in Phoenix & don't want to consider a full re-write. We know that approach [often turns into a quagmire](https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/) and so whatever we choose, we want to be able to adopt it incrementally with our existing codebase (sorry [Nuxt.js](https://vuejs.org), [Svelte](https://svelte.dev) and [Livewire](https://laravel-livewire.com)!)
```

Comment:
*The `Nuxt.js` link links directly to the Vue library itself and not to https://nuxtjs.org/

Reply:
Fixed in ea581c0!